### PR TITLE
ref: Add typehints

### PR DESCRIPTION
**Pull request by untitaker**

### CODE REVIEW

**Comment by mitsuhiko** [comment]
Oh god it's so ugly

File: sentry_sdk/client.py [hunk 1]
```diff
@@ -132,11 +147,12 @@ def _prepare_event(self, event, hint, scope):
                 new_event = before_send(event, hint)
             if new_event is None:
                 logger.info("before send dropped event (%s)", event)
-            event = new_event
+                event = new_event  # type: ignore
```
Comment (untitaker) [inline sentry_sdk/client.py]: oof

File: sentry_sdk/api.py [hunk 2]
```diff
@@ -35,16 +36,20 @@ def capture_event(event, hint=None):
 
 @hubmethod
 def capture_message(message, level=None):
+    # type: (str, Optional[Any]) -> Optional[str]
     hub = Hub.current
     if hub is not None:
         return hub.capture_message(message, level)
+    return None
```
Comment (mitsuhiko) [inline sentry_sdk/api.py]: is this `return None` mess really necessary?

File: sentry_sdk/hub.py [hunk 3]
```diff
@@ -163,18 +188,24 @@ def run(self, callback):
...
         """Returns the integration for this hub by name or class.  If there
         is no client bound or the client does not have that integration
         then `None` is returned.
 
         If the return value is not `None` the hub is guaranteed to have a
         client attached.
         """
-        if not isinstance(name_or_class, string_types):
-            name_or_class = name_or_class.identifier
+        if isinstance(name_or_class, str):
```
Comment (mitsuhiko) [inline sentry_sdk/hub.py]: Should this not be `string_types` for 2.x?

File: sentry_sdk/utils.py [hunk 4]
```diff
@@ -209,39 +233,47 @@ def should_hide_frame(frame):
...
+    filename,  # type: str
+    lineno,  # type: int
+    loader=None,  # type: SourceFileLoader
+    module=None,  # type: str
+):
+    # type: (...) -> Tuple[List[str], Optional[str], List[str]]
     context_lines = 5
     source = None
-    if loader is not None and hasattr(loader, "get_source"):
+    if loader is not None and module is not None and hasattr(loader, "get_source"):
```
Comment (mitsuhiko) [inline sentry_sdk/utils.py]: Sure `module` is necessary? I think some loaders are module specific and ignore the first argument to `get_source`.

File: sentry_sdk/hub.py (hunk 3 above)
Comment (untitaker) [reply sentry_sdk/hub.py]: imo an integration identifier really always should be a native string, and this way mypy is able to infer that name_or_class can later only be one of the two union variants

File: sentry_sdk/utils.py (hunk 4 above)
Comment (untitaker) [reply sentry_sdk/utils.py]: I reverted it, typehints on stdlib are broken

File: sentry_sdk/api.py (hunk 2 above)
Comment (untitaker) [reply sentry_sdk/api.py]: https://github.com/python/mypy/issues/4123

**Comment by untitaker** [comment]
ref https://github.com/PyCQA/pyflakes/issues/422

**Pull request merged**